services/apis/nondbgapUsersync/nonDbgapTasks.js

```diff
@@ -0,0 +1,139 @@
+    I.fillField('USER', process.env.RAS_TEST_USER_1_USERNAME);
+    I.fillField('PASSWORD', process.env.RAS_TEST_USER_1_PASSWORD);
+    I.waitForElement({ xpath: 'xpath: //button[contains(text(), \'Sign in\')]' });
+    I.click({ xpath: 'xpath: //button[contains(text(), \'Sign in\')]' });
+    I.saveScreenshot('AfterSignIn.png');
+    // checking if the authcode is present in the url after login
+    I.seeInCurrentURL('code');
+    // grabbing the authcode from the url and storing it in the variable
+    const authCodeURL = await I.grabCurrentUrl();
+    const url = new URL(authCodeURL);
+    const authCode = url.searchParams.get('code');
+    // check if the authCode isnt empty
+    expect(authCode).to.not.to.be.empty;
+    const code = authCode;
+    console.log(`Successfully retrieved the code from URL : ${code}`);
+
+    // getting access_token from authCode
+    console.log('Retrieving access_token ...');
+    const data = queryString.stringify({
+      grant_type: 'authorization_code',
+      code: `${code}`,
+      client_id: `${I.cache.RAS_clientID}`,
+      client_secret: `${I.cache.RAS_secretID}`,
+      scope: `${scope}`,
+      redirect_uri: `https://${TARGET_ENVIRONMENT}/user`,
+    });
+    // sending request
+    const getRASTokenReq = await I.sendPostRequest(
+      authURL,
+      data,
+      {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    );
+    if (getRASTokenReq.status === 200) {
+      console.log('Retrieved access_token');
+      I.cache.RAS_ACCESS_TOKEN = getRASTokenReq.data.access_token;
+    }
+    return I.cache.RAS_ACCESS_TOKEN;
+  },
+
+  async checkDbGapAccess(token) {
+    // checking the user access to DbGap projects and try sending presigned url requests
+    let id = null;
+    console.log('### Getting the DBGap project access ...');
+    const userDBGapProjectResp = await I.sendGetRequest(
+      `https://${TARGET_ENVIRONMENT}/user/user`,
+      { Authorization: `bearer ${token}` },
+    ).then((res) => new Gen3Response(res));
+
+    const dbGapProjectAccess = userDBGapProjectResp.body.project_access;
+    const projects = Object.keys(dbGapProjectAccess)[0];
+    const projectId = projects[Math.floor(Math.random() * projects.length)];
+    // get index records for the acl project
+    const indexdRecordDID = await I.sendGetRequest(
+      `https://${TARGET_ENVIRONMENT}/index/index?acl=${projectId}`,
+      { Authorization: `bearer ${token}` },
+    );
+    // select the first indexd record
+    if ((indexdRecordDID.data.records.length) > 0) {
+      id = indexdRecordDID.data.records[0].id;
+    }
+    // return the id
+    return id;
+  },
+
+  async presignedURLRequest(fence, project, projectFile, user, token) {
+    console.log(`Creating presigned url with mainAcct user for ${project} after running usersync`);
+    const signedURLs3 = await fence.do.createSignedUrl(projectFile, ['protocol=s3'], token);
+    if (signedURLs3.data.status === 200) {
```

Comment:
To my knowledge, to get status code from a response should use `signedURLs3.status`. response.data will return the response body.